src/core/src/op/avg_pool.cpp

```diff
@@ -19,31 +19,12 @@ AvgPool::AvgPool(const Output<Node>& arg,
                  const Shape& pads_end,
                  const Shape& kernel,
                  bool exclude_pad,
-                 op::RoundingType rounding_type,
-                 const PadType& auto_pad)
-    : Op({arg}),
-      m_kernel(kernel),
-      m_strides(strides),
-      m_pads_begin(pads_begin),
-      m_pads_end(pads_end),
-      m_exclude_pad(exclude_pad),
-      m_auto_pad(auto_pad),
-      m_rounding_type(rounding_type) {
+                 RoundingType rounding_type,
+                 PadType auto_pad)
+    : util::AvgPoolBase(arg, strides, pads_begin, pads_end, kernel, exclude_pad, rounding_type, auto_pad) {
     constructor_validate_and_infer_types();
```

Comment:
As `constructor_validate_and_infer_types` is called on base class creation and called by this class, please double check if this function is called only once per class (there is no double call of this function for base class).